ci/chapter.rst

```diff
@@ -19,86 +21,98 @@ the repository.
+This design will let us handle hardware failures on the fly by enabling a distributed architecture. We can have the observer run on one machine, the test job dispatcher on another, and the test runners on another, and they can all communicate with each other over a network. If any of these machines go down, we can schedule a new machine to go up on the network, so the system becomes fail-safe. 
+
+This project does not include auto-recovery code, as that is dependent on your distributed system's architecture, but in the real world, CI systems are run in a distributed environment like this so they can have failover redundancy (i.e., we can fall back to a standby machine if one of the machines a process was running on becomes defunct).
 
 For the purposes of this project, each of these processes will run locally, and you must kick them off individually. Since the processes need to communicate with each other, they will run on distinct local ports.
 
-Files in This Project
+Files in this Project
 ---------------------
 
-This project contains python files for each of these components: the repository observer (repo_observer.py), the test job dispatcher (dispatcher.py), and the test runner (test_runner.py). Each of these three processes communicate with each other using sockets, and since the code used to transmit information is shared by all of them, there is a helpers.py file that contains it, so each process imports the 'communicate' function from here instead of having it duplicated in the file.
+This project contains Python files for each of these components: the repository observer (`repo_observer.py`), the test job dispatcher (`dispatcher.py`), and the test runner (`test_runner.py`). Each of these three processes communicate with each other using sockets, and since the code used to transmit information is shared by all of them, there is a `helpers.py` file that contains it, so each process imports the `communicate` function from here instead of having it duplicated in the file.
 
-There are also bash script files used by these processes. These script files are used to execute bash and git commands in an easier way than constantly using python's operating system-level modules like 'os' and 'subprocess'.
+There are also bash script files used by these processes. These script files are used to execute bash and git commands in an easier way than constantly using Python's operating system-level modules like `os` and `subprocess`.
```

Comment:
I changed a few formatting things, probably changing RST to Markdown. We need chapters to be in Markdown for the final book production process.